internal/cmd/grpc.go

```diff
@@ -323,12 +323,12 @@ func NewGRPCServer(
 	var (
 		// authOpts is a slice of options that will be passed to the authentication service.
 		// it's initialized with the default option of skipping authentication for the health service which should never require authentication.
-		authOpts = []containers.Option[auth.InterceptorOptions]{
-			auth.WithServerSkipsAuthentication(healthsrv),
+		authOpts = []containers.Option[authmiddlewaregrpc.InterceptorOptions]{
+			authmiddlewaregrpc.WithServerSkipsAuthentication(healthsrv),
 		}
 		skipAuthIfExcluded = func(server any, excluded bool) {
 			if excluded {
-				authOpts = append(authOpts, auth.WithServerSkipsAuthentication(server))
+				authOpts = append(authOpts, authmiddlewaregrpc.WithServerSkipsAuthentication(server))
```

Comment:
As each service now has a method, we could move this logic into those methods.
We just need to get this configuration or even just a flag onto the server structs themselves.
Then they can return that boolean.

Reply:
I see you have a TODO on the auth middleware itself that I think is eluding to this.
Could defer to a later PR 👍

Reply:
yeah i left this here but could do it for most of the other servers as can use their config like you said. the only thing we need to do is fix the healthcheck server (from GRPC) to implement these new methods which would mean wrapping it in server code we control